include/lib.h

```diff
@@ -144,11 +145,42 @@ static inline uint64_t rdmsr(uint32_t msr_idx) {
     return (((uint64_t) high) << 32) | low;
 }
 
+/*
+ * read the msr_idx msr; the result is valid iff *read is true
+ */
+static inline uint64_t rdmsr_safe(uint32_t msr_idx, bool *read) {
```

Comment:
Please make this and the following function a no inline functions. Otherwise there will be more exceptions registered than necessary.